cdap-ui/app/cdap/components/PipelineConfigurations/Store/index.js

```diff
@@ -104,12 +110,34 @@ const getCustomConfigFromProperties = (properties) => {
   return customConfig;
 };
 
+const getCustomConfigForDisplay = (properties, engine) => {
+  let currentCustomConfig = getCustomConfigFromProperties(properties);
+  let customConfigForDisplay = {};
+  for (let key in currentCustomConfig) {
+    if (currentCustomConfig.hasOwnProperty(key)) {
+      let newKey = key;
+      if (engine === 'mapreduce' && key.startsWith('system.mapreduce.')) {
```

Comment:
perhaps extract the startsWith to a constant, then in the `slice` method, you can simply call `.length` of the constant